### PR TITLE
Add sanitizeName for sapi consistency

### PR DESCRIPTION
**Pull request by pivnicek**
needed for https://github.com/keboola/db-extractor-common/issues/38

### CODE REVIEW

**Review by tomasfejfar — CHANGES_REQUESTED** [review]
Easy enough to test, so it should have tests IMO

File: src/Keboola/Utils/sanitizeName.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Keboola\Utils;
+
+/**
+ * https://github.com/nette/nette/blob/master/Nette/Utils/Strings.php#L167
+ * Converts to ASCII.
+ * @param  string  UTF-8 encoding
+ * @return string  ASCII
+ */
+function toAscii($s)
```
Comment (tomasfejfar) [inline src/Keboola/Utils/sanitizeName.php]: should be one function per file

**Review by tomasfejfar — CHANGES_REQUESTED** [review]
Easy enough to test, so it should have tests IMO

**Comment by pivnicek** [comment]
Good points, thanks!

**Comment by pivnicek** [comment]
I think it's better now.  thanks for eyes @tomasfejfar

File: tests/Keboola/Utils/SanitizeNameTest.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Keboola\Utils;
+
+class SanitizeNameTest extends \PHPUnit_Framework_TestCase
+{
+
+    public function testSanitizeName()
```
Comment (tomasfejfar) [inline tests/Keboola/Utils/SanitizeNameTest.php]: dataprovider pls :)

Comment (tomasfejfar) [reply tests/Keboola/Utils/SanitizeNameTest.php]: also test some value with utf8 characters (👍 or some Vietnamese, which could be real use case)

File: tests/Keboola/Utils/ToAsciiTest.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace Keboola\Utils;
+
+class ToAsciiTest extends \PHPUnit_Framework_TestCase
+{
+    public function testToAscii()
```
Comment (tomasfejfar) [inline tests/Keboola/Utils/ToAsciiTest.php]: data povider too

**Review by tomasfejfar — COMMENTED** [review]
Good enough like this, but I think it might be improved.

**Comment by pivnicek** [comment]
I ended up changing it to `sanitizeColumnName` because I saw it was some missing logic that was here:
https://github.com/keboola/connection/blob/master/library/Keboola/Filter/ColumnNameFilter.php

dataProviders are there, natural great idea (these other tests here could use them too:)

**Comment by pivnicek** [comment]
got to be careful here with the system differences locally versus travis.  I get different test results locally.  we should dockerize it.  But I don't want to spend much more time here than necessary.

**Comment by pivnicek** [comment]
tak @tomasfejfar muzu to mergnout?

File: tests/Keboola/Utils/SanitizeColumnNameTest.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public function namesToSanitize()
+    {
+        return [
+            [
+                "_~dlažební  %_kostky_~",
+                "dlazebni__kostky"
+            ],[
+                "test-vn-đá cuội",
```
Comment (tomasfejfar) [inline tests/Keboola/Utils/SanitizeColumnNameTest.php:23]: This does not seem to be a correct behavior IMO
If you put "test-*" it should output "test-" (if rest of the characters are invalid), no?
dtto with the jp.

File: tests/Keboola/Utils/ToAsciiTest.php [hunk 5]
```diff
@@ -4,9 +4,28 @@
...
+    }
+
+    public function testStrings()
+    {
+        return [
+            [
+                "_~dlažební  %_kostky_~",
+                "_~dlazebni  %_kostky_~"
+            ],[
+                "test-vn-đá cuội",
```
Comment (tomasfejfar) [inline tests/Keboola/Utils/ToAsciiTest.php:23]: here again - should output "test-" imho

**Review by tomasfejfar — COMMENTED** [review]
Maybe I understand it wrong, but still not happy about it, sorry :/

**Review by tomasfejfar — APPROVED** [review]

File: tests/Keboola/Utils/SanitizeColumnNameTest.php (hunk 4 above)
Comment (ondrejhlavacek) [reply tests/Keboola/Utils/SanitizeColumnNameTest.php:23]: a simple test suite that shows some of the caveats is here - https://github.com/keboola/kbc-ui/blob/master/src/scripts/utils/string.spec.js

**Pull request merged**